### PR TITLE
Updated docstrings for utilities.iterables

### PR DESCRIPTION
**Pull request by oscargus**
#### Brief description of what is fixed or changed
Happened to stumble upon a few formatting issues in the docstring for utilities.iterables so made an attempt to fix what I saw.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/utilities/iterables.py [hunk 1]
```diff
@@ -2034,18 +2043,21 @@ def generate_oriented_forest(n):
...
     ========
 
     >>> from sympy.utilities.iterables import generate_oriented_forest
     >>> list(generate_oriented_forest(4))
     [[0, 1, 2, 3], [0, 1, 2, 2], [0, 1, 2, 1], [0, 1, 2, 0], \
     [0, 1, 1, 1], [0, 1, 1, 0], [0, 1, 0, 1], [0, 1, 0, 0], [0, 0, 0, 0]]
+
+    References
+    ==========
+    .. [1] T. Beyer and S.M. Hedetniemi: constant time generation of \
```
Comment (sidhantnagpal) [inline sympy/utilities/iterables.py]: A newline before `.. [1]` could be added. Also, the backslash (indicating a continued line) can be removed.

File: sympy/utilities/iterables.py [hunk 2]
```diff
@@ -1443,7 +1446,7 @@ def partitions(n, m=None, k=None, size=False):
 
     Reference:
```
Comment (sidhantnagpal) [inline sympy/utilities/iterables.py]: This section could also follow the Sphinx format.

**Comment by smichr** [comment]
```
Warning, treated as error:
/home/travis/build/sympy/sympy/sympy/utilities/iterables.py:docstring of sympy.utilities.iterables.subsets:10:Error in "rubric" directive:
no content permitted.
.. rubric:: Examples
       >>> from sympy.utilities.iterables import subsets
       ``subsets(seq, k)`` will return the `
Makefile:49: recipe for target 'html' failed
make: *** [html] Error 2
```

File: sympy/utilities/iterables.py [hunk 3]
```diff
@@ -476,19 +476,19 @@ def variations(seq, n, repetition=False):
...
+       whereas there are `2^n` subsets all together. If `k` is ``None`` then all
+       `2^n` subsets will be returned from shortest to longest.
 
        Examples
        ========
 
        >>> from sympy.utilities.iterables import subsets
 
-       subsets(seq, k) will return the n!/k!/(n - k)! k-subsets (combinations)
+       ``subsets(seq, k)`` will return the `\frac{n!}{k!(n - k)!}` `k`-subsets (combinations)
```
Comment (smichr) [inline sympy/utilities/iterables.py]: maybe it doesn't like formatted text in the examples region when it is parsing for output?

**Comment by oscargus** [comment]
I've updated some stuff. Maybe the problem was that the line started with a double back-tick as it can be used in other parts of the Example section.

**Comment by oscargus** [comment]
I have no idea what is wrong here...

**Comment by smichr** [comment]
> Maybe the problem was that the line started with a double back-tick

You could start the sentence as 'By default, ``...'

**Comment by jksuom** [comment]
I suspect this is the classic case of missing `r` from the doctring. There are now new escape sequences.

**Comment by oscargus** [comment]
@smichr In the second from last PR I tried that.

@jksoum I escaped in the Latex commands last time.

Weird thing is that it complains at exactly the same position despite these two things being different between the runs. I was hoping that it would complain on the next unescaped Latex command but no.

Thanks for the suggestions anyway!

**Comment by oscargus** [comment]
New attempt with both of the suggestions. Primarily to see if the errors move.

I guess one way to a solution is to make sure that I can build it locally...

**Comment by jksuom** [comment]
I would put `r` on the docstring of `variations`. That is where the message points to.

File: sympy/utilities/iterables.py [hunk 4]
```diff
@@ -500,7 +500,7 @@ def subsets(seq, k=None, repetition=False):
            [(1, 2), (1, 3), (2, 3)]
 
 
-       subsets(seq, k, repetition=True) will return the (n - 1 + k)!/k!/(n - 1)!
+       ``subsets(seq, k, repetition=True)`` will return the `\\frac{(n - 1 + k)!}{k!(n - 1)!}`
```
Comment (sidhantnagpal) [inline sympy/utilities/iterables.py]: Should this (and the similar instance above) be \`\frac ...\`, now that the docstring is prefixed by `r"""`?

File: sympy/utilities/iterables.py [hunk 5]
```diff
@@ -2034,18 +2060,21 @@ def generate_oriented_forest(n):
...
     ========
 
     >>> from sympy.utilities.iterables import generate_oriented_forest
     >>> list(generate_oriented_forest(4))
     [[0, 1, 2, 3], [0, 1, 2, 2], [0, 1, 2, 1], [0, 1, 2, 0], \
     [0, 1, 1, 1], [0, 1, 1, 0], [0, 1, 0, 1], [0, 1, 0, 0], [0, 0, 0, 0]]
+
+    References
+    ==========
+    .. [1] T. Beyer and S.M. Hedetniemi: constant time generation of
```
Comment (sidhantnagpal) [inline sympy/utilities/iterables.py]: See the first part of https://github.com/sympy/sympy/pull/16143#discussion_r261837584.

**Comment by sidhantnagpal** [comment]
Except for a few minor suggestions, this looks good to me.

**Pull request merged**